src/miradorDownloadPlugin.js

```diff
@@ -1,17 +1,107 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import ListItemText from '@material-ui/core/ListItemText';
+import MenuItem from '@material-ui/core/MenuItem';
+import Typography from '@material-ui/core/Typography';
+import DownloadIcon from '@material-ui/icons/VerticalAlignBottomSharp';
+import { getCanvasLabel, getSelectedCanvases } from 'mirador/dist/es/src/state/selectors/canvases';
+import CanvasDownloadLinks from './CanvasDownloadLinks';
+
+const mapStateToProps = (state, { windowId }) => ({
+  canvases: getSelectedCanvases(state, { windowId }),
+  canvasLabel: canvasIndex => (getCanvasLabel(state, { canvasIndex, windowId })),
+});
 
 class MiradorDownload extends Component {
+  static fullImageUrlForCanvas(canvas) {
+    return `${canvas.getCanonicalImageUri().replace(/\/full\/.*\/0\//, '/full/full/0/')}?download=true`;
+  }
+
+  static thousandPixelWideImageForCanvas(canvas) {
+    return `${canvas.getCanonicalImageUri('1000')}?download=true`;
+  }
+
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      modalDisplayed: false,
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+    this.handleDialogClose = this.handleDialogClose.bind(this);
+  }
+
+  handleClick() {
+    const { modalDisplayed } = this.state;
+    this.setState({ modalDisplayed: !modalDisplayed });
+  }
+
+  handleDialogClose() {
+    this.setState({ modalDisplayed: false });
+  }
+
   render() {
+    const { canvases, canvasLabel } = this.props;
+    const { modalDisplayed } = this.state;
     return (
       <div>
-        <h2>Download</h2>
+        <MenuItem onClick={this.handleClick}>
+          <DownloadIcon />
+          <ListItemText inset primaryTypographyProps={{ variant: 'body1' }}>
+            Download
+          </ListItemText>
+        </MenuItem>
+        <Dialog
+          disableEnforceFocus
+          onClose={this.handleDialogClose}
+          open={modalDisplayed}
+          scroll="paper"
+        >
+          <DialogTitle disableTypography>
+            <Typography variant="h2">Download</Typography>
+          </DialogTitle>
+          <DialogContent>
+            {canvases.map(canvas => (
+              <CanvasDownloadLinks
```

Comment:
There's an awful lot of canvas properties getting passed here; I wonder if it'd be better to pass the whole canvas object instead (and, that'd let us move those image uri properties into `CanvasDownloadLinks`)? 🤷‍♂

Reply:
I _think_ the concern was that passing the canvas object would cause re-renders when we didn't want (because JS object comparison) but maybe passing the `canvas.id` as the key would prevent that?

Reply:
That makes sense, but if the canvas object is changing, won't `MiradorDownload` re-render anyway?

Reply:
In its current implementation, yes, that is true.  If the component were to ever get moved to a panel/companion-window (or something like that), that would all need to change (but I supposed I don't need to concern myself w/ that now)

Reply:
Updated to just pass the `canvas` and `canvasLabel` (`canvasLabel` is connected to the redux store and I either had to pass the function or the value, and it felt like it made more sense to just pass the value)